runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -396,7 +396,8 @@ protected long getMaxGlobalFromMaxEntries(int maxEntries) {
 
         long size = streamAddressSpace.size();
         if (size == 0L) {
-            return Address.NON_ADDRESS;
+            // The trim mark will allow to detect trimmed exceptions if the seeked address falls behind it
```

Comment:
Can you also add a unit test that targets  `IStreamView` since this is a separate bug altogether.

Reply:
Done, added 4 test cases in StreamViewTest.

BTW, the IStreamView APIs that are used to directly consume the stream (not the ones used through VLO) do not work for checkpointable streams. I remember Wenbin was going to work on this some time back but then he got deviated to something else. But this is something important that we should address any time soon. I'll add it as an issue.